### ISSUE TITLE
Base table or view already exists: 1050 Table 'telescope_entries' already exists

### ISSUE THREAD

**Issue by SlyDave** [VemtoOrg/vemto2-issues#143]
**Error Message:**
schema-reader-db Error: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'telescope_entries' already exists (Connection: mysql, SQL: create table `telescope_entries` (`sequence` bigint unsigned not null auto_increment primary key, `uuid` char(36) not null, `batch_id` char(36) not null, `family_hash` varchar(255) null, `should_display_on_index` tinyint(1) not null default '1', `type` varchar(20) not null, `content` longtext not null, `created_at` datetime null) default character set utf8mb4 collate 'utf8mb4_0900_ai_ci')

**Stack:**
```
schema-reader-db Error: #0 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Connection.php(767): Illuminate\Database\Connection->runQueryCallback()
#1 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Connection.php(560): Illuminate\Database\Connection->run()
#2 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Schema\Blueprint.php(109): Illuminate\Database\Connection->statement()
#3 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Schema\Builder.php(564): Illuminate\Database\Schema\Blueprint->build()
#4 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Schema\Builder.php(418): Illuminate\Database\Schema\Builder->build()
#5 C:\www\workrest\database\migrations\2018_08_08_100000_create_telescope_entries_table.php(23): Illuminate\Database\Schema\Builder->create()
#6 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(482): Illuminate\Database\Migrations\Migration@anonymous->up()
#7 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(408): Illuminate\Database\Migrations\Migrator->runMethod()
#8 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(417): Illuminate\Database\Migrations\Migrator->Illuminate\Database\Migrations\{closure}()
#9 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(214): Illuminate\Database\Migrations\Migrator->runMigration()
#10 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(749): Illuminate\Database\Migrations\Migrator->Illuminate\Database\Migrations\{closure}()
#11 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(214): Illuminate\Database\Migrations\Migrator->write()
#12 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(179): Illuminate\Database\Migrations\Migrator->runUp()
#13 C:\www\workrest\vendor\laravel\framework\src\Illuminate\Database\Migrations\Migrator.php(122): Illuminate\Database\Migrations\Migrator->runPending()
#14 phar://C:/Program Files/Vemto/resources/static/VMTTL2/index.php(111): Illuminate\Database\Migrations\Migrator->run()
#15 phar://C:/Program Files/Vemto/resources/static/VMTTL2/common/Vemto.php(111): {closure}()
#16 phar://C:/Program Files/Vemto/resources/static/VMTTL2/index.php(18): Vemto\Vemto::execute()
#17 C:\Program Files\Vemto\resources\static\VMTTL2(14): require('...')
#18 {main}
```

---

Vemto 2.0.14
win32 10.0.22631
Locale: en-GB
App version: 2.0.14

Not sure why the Schema Reader is trying to create telescope tables? it should only be read only no?

**Comment by TiagoSilvaPereira** [comment]
Hi @SlyDave 

> Not sure why the Schema Reader is trying to create telescope tables? it should only be read only no?

No, the schema reader needs to create a separate database to read the schema definition. 

Maybe you are trying to migrate a database that is blocked (or an existing database), as Vemto needs to drop the database before recreating it and migrating the tables. Please try changing the database name on Connection Settings a let Vemto create the database by itself:

![image](https://github.com/TiagoSilvaPereira/vemto2-issues/assets/11933789/6cdc91f9-44c8-42ea-86c8-006d5de8d084)

It is important to note that Vemto never uses the same database as the application. It is mandatory that the database name starts with the prefix "vemto_", and even so, Vemto will check if the database configured in the .env is the same and will not allow its use in this case.

Also, you don't need to create this database manually. Vemto will be responsible for creating and migrating the schema definition database.

**Comment by SlyDave** [comment]
Thanks for the quick response.

It seems Vemo isn't running the mysql-schema.sql file from flattening before trying to apply migrations to the copy of the database it creates?

I deleted migrate "C:\www\workrest\database\migrations\2018_08_08_100000_create_telescope_entries_table.php" which was a migration that shouldn't have existed (an artefact of my Laravel 10 to 11 upgrade)

But now it's erroring on the next migration, stating a table doesn't exist that most certainly does - in the flattened schema at least.

mysql-schema.sql
```
DROP TABLE IF EXISTS `contracts`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `contracts` (
```

2022_09_08_133405_drop_assignment_id_from_contracts.php
` Schema::table('contracts', static function (Blueprint $table) {`

```
schema-reader-db Error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vemto_schema_reader.contracts' doesn't exist (Connection: vemto_db_connection, SQL: alter table `contracts` add `user_id` int unsigned not null after `id`, add `vessel_id` int unsigned not null after `user_id`)
```

**Comment by TiagoSilvaPereira** [comment]
> Thanks for the quick response.
> 
> It seems Vemo isn't running the mysql-schema.sql file from flattening before trying to apply migrations to the copy of the database it creates?
> 
> I deleted migrate "C:\www\workrest\database\migrations\2018_08_08_100000_create_telescope_entries_table.php" which was a migration that shouldn't have existed (an artefact of my Laravel 10 to 11 upgrade)
> 
> But now it's erroring on the next migration, stating a table doesn't exist that most certainly does - in the flattened schema at least.
> 
> mysql-schema.sql
> 
> ```
> DROP TABLE IF EXISTS `contracts`;
> /*!40101 SET @saved_cs_client     = @@character_set_client */;
> /*!50503 SET character_set_client = utf8mb4 */;
> CREATE TABLE `contracts` (
> ```
> 
> 2022_09_08_133405_drop_assignment_id_from_contracts.php ` Schema::table('contracts', static function (Blueprint $table) {`
> 
> ```
> schema-reader-db Error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'vemto_schema_reader.contracts' doesn't exist (Connection: vemto_db_connection, SQL: alter table `contracts` add `user_id` int unsigned not null after `id`, add `vessel_id` int unsigned not null after `user_id`)
> ```

hmmm I'll check this, I did not test with mysql-schema.sql yet. This should work, because Vemto just runs "php artisan migrate" behind the scenes, but maybe I forgot something. I'll check this, thanks!!

**Comment by TiagoSilvaPereira** [comment]
Fixed on 2.0.24
